website/end_loader.py

```diff
@@ -0,0 +1,39 @@
+# end_loader runs once all main application code is done.
+from browser import document, window
+def end_loader(done_message="Ready!"):
+
+    def finished_ready(event):
+        print("finished ready!")
+        # Loading should also be done with its gradient animation too. We can hide it now
+        document["loading"].style.display = "none"
+        document["loading"].style.cursor = "default"
+        document['psychology'].unbind("animationend")
+        del document["loading"]
+
+
+    def finished_loading(event):
+        """Finish the loading animation"""
+        print("Finished animating!")
+
+        document["loading_icon"].classList.remove('fa-circle-o-notch')
+        document["loading_icon"].classList.remove('fa-spin')
+        document["loading_icon"].classList.add("fa-thumbs-o-up")
+
+        # We set loading to the gradientout class which animates from grey background to white
+        document["loading"].classList.add("gradientout")
+        # We set psychology to the growout class which has a single animation.
+        document["psychology"].classList.add("growout")
+        document["psychology"].classList.remove("pulsing")
+        document["loading_message"].innerHTML = done_message
```

Comment:
Suggest to add output to console. This way, the console can naturally become a profiler for the developers to know which part took how long.
```suggestion
        document["loading_message"].innerHTML = done_message
        print(done_message)
```

And, such line brings me to this finding. There is roughly 1 second gap between this line (which is "Let's go") and the [line 6](https://github.com/rayluo/brython-project-template/pull/2/files#diff-6b9c6ba2f92b2bf7617a5c4de15964851cc20c2c846014c0ea10c1d3e5a998a1R6) "finished ready". This brings an important topic for us: **Isn't the animation supposed to be ease the anxiety *during* the waiting, rather than *cause* more waiting time?**

Our work in this PR already brings the non-blocking spinning circle with each message (and somehow you even have the spinning mouse cursor available for *full screen*, which is great!), I'd suggest to remove the ending eye-candy which merely adds one second to the waiting time.

Reply:
Yes, the way it was programmed was that we would wait for the existing pulse animation to end its sequence (pulse animation was set to 2 seconds so this was a max 2 second wait) and then the eye candy would run for 1 second also. So in worse case it added 3 seconds wait. If I remove all that the code is much simplified (we can lose end-loader entirely) but the transitions are very jump-cut. It works faster but it feels (to me) less pleasant an experience. But again, I make the change and let you decide.